src/services/CertificateGenerationService.ts

```diff
@@ -292,14 +292,14 @@ class CertificateGenerationService {
         rawDefects.forEach((defect: any) => {
             switch (defect.deficiencyCategory.toLowerCase()) {
                 case "dangerous":
-                    if (testTypes.testResult === TestResultType.PRS && type === "FAIL_DATA") {
+                    if ((testTypes.testResult === TestResultType.PRS || defect.prs)  && type === "FAIL_DATA") {
```

Comment:
We need two unit tests and corresponding int tests under the context failing test result which is clear and sufficiently understandable to test these two logical checks under the cases
1. "dangerous"
2. "major"